javascript/ql/src/semmle/javascript/dataflow/DataFlow.qll

```diff
@@ -987,6 +987,14 @@ module DataFlow {
     )
   }
 
+  /**
+   * Gets the data flow node representing the destructuring source of definition `def`, taking
+   * flow through IIFE calls into account.
+   */
+  private AST::ValueNode destructuringDefSourceNode(VarDef def) {
```

Comment:
I'm not sure we need a separate predicate for this. The two uses of `defSourceNode/1` (in `localFlowStep` above and in `defSourceNode/2` below) look like they should work with both the destructuring and the non-destructuring case.